File: src/test/java/org/geepawhill/contentment/utility/NamesTest.java
package org.geepawhill.contentment.utility;


import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class NamesTest {

    @BeforeEach
    public void before() {
        Names.INSTANCE.reset();
    }

    @Test
    public void sequences() {
        assertEquals("Test_1", Names.INSTANCE.make("Test"));
        assertEquals("Test_2", Names.INSTANCE.make("Test"));
    }

    @Test
    public void rootCounts() {
        assertEquals("Test_1", Names.INSTANCE.make("Test"));
        assertEquals("Something_1", Names.INSTANCE.make("Something"));
    }

    @Test
    public void className() {
        assertEquals("String_1", Names.INSTANCE.make(String.class));
    }

    @Test
    public void reset() {
        assertEquals("String_1", Names.INSTANCE.make(String.class));
        assertEquals("String_2", Names.INSTANCE.make(String.class));
        Names.INSTANCE.reset();
        assertEquals("String_1", Names.INSTANCE.make(String.class));
    }

}

File: src/test/kotlin/org/geepawhill/contentment/utility/NamesTest.kt
package org.geepawhill.contentment.utility


import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.assertEquals

class NamesTest {

    @BeforeEach
    fun before() {
        Names.reset()
    }

    @Test
    fun sequences() {
        assertEquals("Test_1", Names.make("Test"))
        assertEquals("Test_2", Names.make("Test"))
    }

    @Test
    fun rootCounts() {
        assertEquals("Test_1", Names.make("Test"))
        assertEquals("Something_1", Names.make("Something"))
    }

    @Test
    fun className() {
        assertEquals("String_1", Names.make(String::class.java))
    }

    @Test
    fun reset() {
        assertEquals("String_1", Names.make(String::class.java))
        assertEquals("String_2", Names.make(String::class.java))
        Names.reset()
        assertEquals("String_1", Names.make(String::class.java))
    }

}
